Tests/NIOFileSystemIntegrationTests/XCTestExtensions.swift

```diff
@@ -69,4 +69,16 @@ func XCTAssertThrowsFileSystemErrorAsync<R>(
         }
     }
 }
+
+func XCTAssertNoThrowAsync<T>(
```

Comment:
we shouldn't squat on XCTest's `XCT*` name space, you can make it `NIOAssertNoThrowAsync` or smth